### PR TITLE
Logging sidecar refactoring

### PR DESCRIPTION
**Pull request by Crassirostris**
Moved from https://github.com/kubernetes/kubernetes.github.io/pull/2114

This is a replacement for https://github.com/kubernetes/contrib/tree/master/logging/fluentd-sidecar-gcp and https://github.com/kubernetes/contrib/tree/master/logging/fluentd-sidecar-es, describing general sidecar approach, including things not covered by docs above:

* Configuring sidecar logging agent using ConfigMaps
* Alternative use for sidecar, where it streams logs to `stdout`/`stderr` rather than has a logging agent inside

Related to https://github.com/kubernetes/kubernetes.github.io/issues/548

CC @piosz @devin-donnelly @steveperry-53

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/2141)

### CODE REVIEW

**Comment by piosz** [comment]
cc @kubernetes/sig-instrumentation-misc

File: docs/user-guide/logging/overview.md [hunk 1]
```diff
@@ -91,17 +88,111 @@ Kubernetes doesn't specify a logging agent, but two optional logging agents are
...
 
-The concrete implementation of the logging agent, the interface between agent and the application, and the interface between the logging agent and the logs back-end are completely up to a you. For an example implementation, see the [fluentd sidecar container](https://github.com/kubernetes/contrib/tree/b70447aa59ea14468f4cd349760e45b6a0a9b15d/logging/fluentd-sidecar-gcp) for the Stackdriver logging backend.
+#### Streaming sidecar container
 
-**Note:** Using a sidecar container for logging may lead to significant resource consumption.
+![Sidecar container with a streaming container](/images/docs/user-guide/logging/logging-with-streaming-sidecar.png)
+
+Using this approach you can re-use per-node agent and kubelet log handling
+mechanisms. A separate container should contain simple piece of software that
+reads logs from a file, a socker or the journald and prints it to the
```
Comment (Crassirostris) [inline docs/user-guide/logging/overview.md]: ACK

**Comment by Crassirostris** [comment]
@piosz I had some input from @choused and will add some more information soon. I will mention you when it's ready

**Comment by steveperry-53** [comment]
@devin-donnelly, Would you like me to do the docs review for this PR? It would make sense, because I just finished working on https://github.com/kubernetes/kubernetes.github.io/pull/2142.

@piosz, Now that https://github.com/kubernetes/kubernetes.github.io/pull/2142 is merged, can you resolve the conflicts here?

**Comment by piosz** [comment]
Thanks @steveperry-53 for offering your help!
@Crassirostris please rebase.

**Comment by Crassirostris** [comment]
@steveperry-53 Thanks!

@piosz Sure, will prepare this PR for reviewing by the EOW

**Comment by Crassirostris** [comment]
@piosz @steveperry-53 PTAL

**Comment by steveperry-53** [comment]
@Crassirostris, See [this commit](https://github.com/kubernetes/kubernetes.github.io/pull/2141/commits/5256a219bee01221521b6a2d277def6c7dff4344) for suggested edits to overview.md.

**Comment by steveperry-53** [comment]
lgtm for stackdriver.md

**Comment by Crassirostris** [comment]
@steveperry-53 I agree with all changes, except ones which remote prompts from the shell listings. I think this is a common notation, used everywhere in the docs

**Comment by Crassirostris** [comment]
@steveperry-53 Thanks! PTAL

**Pull request merged**